src/dtls13.c

```diff
@@ -0,0 +1,1968 @@
+/* dtls13.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#ifdef WOLFSSL_DTLS13
+
+#include <wolfssl/error-ssl.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/wolfcrypt/aes.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/types.h>
+#include <wolfssl/wolfcrypt/kdf.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+enum rnDirection {
+    PROTECT = 0,
+    DEPROTECT,
+};
+
+typedef struct Dtls13HandshakeHeader {
+    byte msg_type;
+    byte length[3];
+    byte messageSeq[2];
+    byte fragmentOffset[3];
+    byte fragmentLength[3];
+} Dtls13HandshakeHeader;
+
+typedef struct Dtls13RecordPlaintextHeader {
+    byte contentType;
+    ProtocolVersion legacyVersionRecord;
+    byte epoch[2];
+    byte sequenceNumber[6];
+    byte length[2];
+} Dtls13RecordPlaintextHeader;
+
+/* No CID, 16bit Seq number, Length present */
+typedef struct Dtls13RecordCiphertextHeader {
+    /* 0 0 1 C S L E E */
+    byte unifiedHdrFlags;
+    byte sequenceNumber[2];
+    byte length[2];
+} Dtls13RecordCiphertextHeader;
+
+#define MAX_SEQ_SIZE 2
+#define LEN_SIZE 2
+#define RN_MASK_SIZE 16
+#define HEADER_FLAGS_SIZE 1
+#define SEQ_16_LEN 2
+#define SEQ_8_LEN 1
+#define SEQ_FULL_SIZE 4
```

Comment:
Add a `DTLS13_` prefix because they look very close to asn defines ex: `MAX_SEQ_SZ` or `MAX_LENGTH_SZ`.